crypto/keyring/keyring.go

```diff
@@ -122,14 +127,33 @@ type Importer interface {
 	ImportPubKey(uid string, armor string) error
 }
 
+// TODO try to remove to see where it is used
 // LegacyInfoImporter is implemented by key stores that support import of Info types.
 type LegacyInfoImporter interface {
 	// ImportInfo import a keyring.Info into the current keyring.
 	// It is used to migrate multisig, ledger, and public key Info structure.
-	ImportInfo(oldInfo Info) error
+	ImportInfo(oldInfo LegacyInfo) error
+}
+
+type Migrator interface {
+	MigrateAll() (bool, error)
+	Migrate(key string) (bool, error)
+}
+
+// used in migration_test.go
+type Setter interface {
+	SetItem(item keyring.Item) error
+}
+
+// TODO if it is onlyfo tests it should not bei n public interface
+// used in migration_test.go
+type Marshaler interface {
+	ProtoMarshalRecord(k *Record) ([]byte, error)
+	ProtoUnmarshalRecord(bz []byte) (*Record, error)
 }
```

Comment:
Renaming the package might cause cycle dep issue. I am not sure if it is a good good solution. Another possible solution might be to make `keystore.db` field public, so the `keyring_test` package can access it.
```

Reply:
I encountered cycle dep issue due to importing `simapp`

Running tool: /usr/local/go/bin/go test -timeout 120s -run ^TestNewKeyring$ github.com/cosmos/cosmos-sdk/crypto/keyring -tags='ledger test_ledger_mock norace experimental'
```
# github.com/cosmos/cosmos-sdk/crypto/keyring
package github.com/cosmos/cosmos-sdk/crypto/keyring (test)
	imports github.com/cosmos/cosmos-sdk/simapp
	imports github.com/cosmos/cosmos-sdk/baseapp
	imports github.com/cosmos/cosmos-sdk/client
	imports github.com/cosmos/cosmos-sdk/client/flags
	imports github.com/cosmos/cosmos-sdk/crypto/keyring: import cycle not allowed in test
FAIL	github.com/cosmos/cosmos-sdk/crypto/keyring [setup failed]
FAIL
```

Reply:
Since I import `simapp` only to get codec, i can implement `getCodec()` function to avoid this import and cycle dep issue.
```
func getCodec() codec.Codec {
  registry := codectypes.NewInterfaceRegistry()
 cryptocodec.RegisterInterfaces(registry)
 return codec.NewProtoCodec(registry)
}

```

What do you think, @amaury?

Reply:
Yes, that sounds like a good idea!